stages/test/test_dmverity.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+
+import pytest  # type: ignore
+
+from osbuild import testutil
+from osbuild.testutil import has_executable
+
+STAGE_NAME = "org.osbuild.dmverity"
+
+
+@pytest.mark.parametrize("test_data,expected_err", [
+    # bad
+    ({}, "'root_hash_file' is a required property"),
+    ({"root_hash_file": 123}, "123 is not of type 'string'"),
+    # good
+    ({"root_hash_file": "abc"}, ""),
+])
+def test_schema_validation_dmverity(stage_schema, test_data, expected_err):
+    test_input = {
+        "type": STAGE_NAME,
+        "devices": {
+            "data_device": {
+                "path": "some-path",
+            },
+            "hash_device": {
+                "path": "some-path",
+            },
+        },
+        "options": {
+        }
+    }
+    test_input["options"].update(test_data)
+    res = stage_schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="root only")
+@pytest.mark.skipif(not has_executable("mkfs.ext4"), reason="need mkfs.ext4")
+@pytest.mark.skipif(not has_executable("veritysetup"), reason="need veritysetup")
+def test_dmverity_integration(tmp_path, stage_module):
+    fake_dev_path = tmp_path / "dev"
+    fake_dev_path.mkdir()
+
+    fake_data_disk = "xxd1"
+    fake_hash_disk = "xxd2"
+    for fname in [fake_data_disk, fake_hash_disk]:
+        p = fake_dev_path / fname
+        p.write_bytes(b"")
+        os.truncate(p, 10 * 1024 * 1024)
+    subprocess.run(
```

Comment:
This format is not really necessary, dmverity works on the block level so it does not matter if the (fake) device is formatted or not (but if you prefer to keep it that is fine, please add a comment in this case so that our future selfs know it's not strictly needed but it was added other reasons, e.g. `# format is not strictly needed as dmvertify is working on the block level but this makes the test more realistic` (assuming that is the rational)) but honetly I think it's not really needed.